src/AccessToken.php

```diff
@@ -317,4 +395,30 @@ protected function callJwtStatic($method, array $args = [])
             : 'JWT';
         return call_user_func_array([$class, $method], $args);
     }
+
+    /**
+     * Provide a hook to mock calls to the JWT static methods.
+     *
+     * @param string $method
+     * @param array $args
+     * @return mixed
+     */
+    protected function callSimpleJwtDecode(array $args = [])
+    {
+        return call_user_func_array([SimpleJWT::class, 'decode'], $args);
+    }
+
+    /**
+     * Generate a cache key based on the cert location using sha1 with the
+     * exception of using "federated_signon_certs_v3" to preserve BC.
+     *
+     * @param string $certsLocation
+     * @return string
+     */
+    private function getCacheKeyFromCertLocation($certsLocation)
+    {
+        return $certsLocation === self::FEDERATED_SIGNON_CERT_URL
+            ? 'federated_signon_certs_v3'
+            : sha1($certsLocation);
```

Comment:
perhaps we can prefix this with a readable string to give a hint as to what it means for anyone who may be inspecting the cache? might be good for the default ones too.

```php
        $key = $certsLocation === self::FEDERATED_SIGNON_CERT_URL
            ? 'federated_signon_certs_v3'
            : sha1($certsLocation);

		return 'google_auth_certs_cache:' . $key;
```

Reply:
Great idea, and done!